src/Metrics.php

```diff
@@ -0,0 +1,182 @@
+            )
+        ) );
+
+        $path = sprintf( '%s/%s', self::METRICS, self::TIMELINE );
+
+        return $this->v1Request($path, $params );
+
+    }
+
+    /**
+     * Returns a batched timeline for one specific type of metric, requires metric ID from your Klaviyo account
+     * @link https://www.klaviyo.com/docs/api/metrics#metric-timeline
+     * @param string $metricID
+     * 6 digit unique identifier of the metric
+     *
+     * @param string $since
+     * To be used with the since argument of the call, can accept UNIX timestamps,
+     * The `since` argument of the call defaults to current time
+     *
+     * @param string $uuid
+     * Can be used with the `since` argument of the call, is obtained via the 'next' attribute of a prior API call.
+     * The `since` argument of the call defaults to current time
+     *
+     * @param int $count defaults to 100,
+     * Number of events to return in a batch
+     *
+     * @param string $sort defaults to 'desc'
+     * Sort order to apply to timeline
+     *
+     * @return bool|mixed
+     *
+     */
+    public function getMetricTimeline( string $metricID, $since = null, $uuid = null, int $count = null, string $sort = null)
+    {
+        $params = $this->setSinceParameter( $since, $uuid );
+
+        $params = $this->filterParams( array_merge(
+            $params,
+            array(
+                self::COUNT => $count,
+                self::SORT => $sort
+            )
+        ) );
+
+        $path = sprintf( '%s/%s/%s', self::METRIC, $metricID, self::TIMELINE );
+
+        return $this->v1Request( $path, $params );
+    }
+
+    /**
+     * Export event data from Klaviyo optionally filtering and segmented on available event properties.
+     * @link https://www.klaviyo.com/docs/api/metrics#metric-export
+     * @param string $metricID
+     * 6 digit unique identifier of the metric
+     *
+     * @param $start_date
+     *
+     * @param $end_date
+     *
+     * @param $unit
+     *
+     * @param $measurement
+     *
+     * @param $where
+     *
+     * @param $by
+     *
+     * @param $count
+     *
+     * @return bool|mixed
+     */
+    public function exportMetricData( $metricID,
+                                      $start_date = null,
+                                      $end_date = null,
+                                      $unit = null,
+                                      $measurement = null,
+                                      $where = null,
+                                      $by = null,
+                                      $count = null )
+    {
+        $params = $this->filterParams(
+            array(
+                self::START_DATE => $start_date,
+                self::END_DATE => $end_date,
+                self::UNIT => $unit,
+                self::MEASUREMENT => $measurement,
+                self::WHERE => $where,
+                self::BY => $by,
+                self::COUNT => $count
+            )
+        );
+
```

Comment:
let's check the params before making a request to ensure someone isn't passing both a where and by parameter. you can only use one according to our docs. probably better to notify the customer. not sure what happens when you attempt to use both but i woudln't be surprised if it's confusing

Reply:
good callout, probably need to do this for all of our SDK's....

Reply:
yup, I'll add a check here